source/component/image_preprocess.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: UTF-8 -*-
+import numpy as np
+import cv2
+from matplotlib import pyplot as plt
+#线性映射数组
+def SubLinear():
+    aSubLinear = [0]*256
+    for i in range(256):
+        if i<60:
+            aSubLinear[i] = int(30.0/60*i)
+        elif i<200:
+            aSubLinear[i] = int((220.0-30.0)/(200-60)*(i-60)+30)
+        else:
+            aSubLinear[i] = int((255.0-220.0)/(255-200)*(i-200)+220)
+    return aSubLinear
+
+#线性变换
+def LinearTrans(image,array=SubLinear()):
+    row=image.shape[0]
+    col=image.shape[1]
+    image2=np.zeros(image.shape,image.dtype)
+    for i in range(row):
+        for j in range(col):
+            pi=image[i][j]
+            image2[i][j]=array[pi]
+    return image2
+
+#图像处理，返回摄像头每一帧处理后的灰度图。
+#img:灰度图
+#zoomR:图像缩放倍数，默认缩小2倍
+#输出处理后的图像
+def imageDW(img,zoomR=2):
```

Comment:
If the zooming size can be a fix number might be better for latter training setting.